synchronization/latch.cc

```diff
@@ -0,0 +1,32 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/synchronization/latch.h"
+
+namespace flutter {
+
+Latch::Latch(size_t count) : initial_(count), count_(count) {}
+
+Latch::~Latch() = default;
+
+void Latch::Wait() {
+  if (count_ > 0) {
+    ftl::MutexLocker lock(&condition_mutex_);
+    while (count_ != 0) {
+      condition_.Wait(&condition_mutex_);
+    }
+  }
+}
+
+void Latch::CountDown() {
+  if (--count_ == 0) {
```

Comment:
I think there's a race here because CountDown() doesn't acquire the condition_mutex.  SignalAll() could therefore run before Latch::Wait() has entered the condition_.Wait()